### PR TITLE
HPCC-32967 Avoid Otel GetGlobalProp spinlock

### PR DESCRIPTION
**Pull request by rpastrana**
- Avoids frequent call to GetGlobalProp
- Adds filescope-wide propegator
- Does not set globalprop

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/jlib/jtrace.cpp [hunk 1]
```diff
@@ -530,6 +530,12 @@ class CTraceManager : implements ITraceManager, public CInterface
 
 static Singleton<CTraceManager> theTraceManager;
 
+// thePropagator can be used be regardless of whether tracing is enabled or not.
+// Can be used to injects Context into or extract it from carriers that travel in-band
+// across process boundaries. Encoding is expected to conform to the HTTP Header Field semantics.
+// Reference this propagator rather than fetching globalpropagator
+nostd::shared_ptr<TextMapPropagator> thePropagator = nostd::shared_ptr<TextMapPropagator>(new opentelemetry::trace::propagation::HttpTraceContext());
```
Comment (ghalliday) [inline system/jlib/jtrace.cpp:537]: trivial: could be static to avoid exporting outside this module.

**Review by ghalliday — APPROVED** [review]
Looks good.  One trivial suggestion, but will merge as is.
Will only merge to 9.6.x and later though, not 9.4.x.

**Comment by rpastrana** [comment]
@ghalliday made the variable static, and cleaned up the comment.
What's keeping us from targeting 9.4?

**Comment by ghalliday** [comment]
Minimizing the impact of the change.  This is not strictly speaking fixing a bug, it is an optimization.  In some ways I don't really like it being included in 9.6.x, but it needs to be because of the systems that need the change, and it should be relatively safe.

**Review by ghalliday — APPROVED** [review]

**Comment by rpastrana** [comment]
> Minimizing the impact of the change. This is not strictly speaking fixing a bug, it is an optimization. In some ways I don't really like it being included in 9.6.x, but it needs to be because of the systems that need the change, and it should be relatively safe.

Fair enough. If load tests using this change show significant performance improvement, would we then consider targeting 9.4 as well?

**Comment by ghalliday** [comment]
I don't think we have any production systems using 9.4.x, and there are many other performance reasons not to be using 9.4.x

**Comment by rpastrana** [comment]
> I don't think we have any production systems using 9.4.x, and there are many other performance reasons not to be using 9.4.x

ok... if that's the case we should note (somewhere) tracing should only be enabled in 9.6.x+

**Pull request merged**